src/extensions/default/HealthData/HealthDataManager.js

```diff
@@ -306,25 +308,41 @@ define(function (require, exports, module) {
     function checkAnalyticsDataSend(event, Eventparams, forceSend) {
         var result         = new $.Deferred(),
             isHDTracking   = prefs.get("healthDataTracking"),
-            isEventDataAlreadySent;
+            isEventDataAlreadySent,
+            url = brackets.config.healthDataServerURL,
+            sendFailed = false;
 
         var options = {
             location: {
                 scope: "default"
             }
         };
 
+        var handleFail = function handleFail() {
+            if(sendFailed) {
+                PreferencesManager.setViewState(Eventparams.eventName, 0, options);
+            } else {
+                sendFailed = true;
+            }
+        };
+
         if (isHDTracking) {
             isEventDataAlreadySent = PreferencesManager.getViewState(Eventparams.eventName);
             PreferencesManager.setViewState(Eventparams.eventName, 1, options);
             if (!isEventDataAlreadySent || forceSend) {
                 sendAnalyticsDataToServer(Eventparams)
-                    .done(function () {
-                        PreferencesManager.setViewState(Eventparams.eventName, 1, options);
-                        result.resolve();
-                    }).fail(function () {
-                        PreferencesManager.setViewState(Eventparams.eventName, 0, options);
-                        result.reject();
+                    .always(function () {
+                        sendAnalyticsDataToServer(Eventparams, url)
+                           .done(function () {
+                               result.resolve();
```

Comment:
Just make sure view state is correct in every case like not able to send health data server but successfully sent on analytics server.

Reply:
View State is unset only if it fails to send Data to both the server. If it fails for any of one server then it will execute else part and sendFailed is assigned to true. when it will fail for second server also It will go in  if part and will unset the view state.